openlibrary/plugins/upstream/code.py

```diff
@@ -42,6 +46,25 @@ def GET(self):
         raise web.seeother(host + '/static' + web.ctx.path)
 
 
+class history(history_api):
```

Comment:
We're not using inheritance in any way, so don't need to inherit that class directly.

```suggestion
class history(delegate.mode):
```

Reply:
I tested with both and I think we do need `history_api` -- if you tested w/ `delegate.mode` and it worked, happy to clean this up and avoid the extra dependencies.

When overwriting an inforgami core, I think we have to extend the class which is currently providing the functionality -- similar to what we did with `core.edit` in `plugins/upstream/code.py`

Reply:
I must have been testing delegate.page instead of delegate.mode; `delegate.mode` seems to work in place of `history_api`